### DISCUSSION TITLE
Checklist for first JOSS submission

### DISCUSSION THREAD

**Discussion by mparno** [MeasureTransport/MParT#139] [General]
I think we're about ready to start preparing a JOSS submission on MParT, but want to make we're all on the same page about what we need to do before submission.    I think the biggest question is whether we want to include MParT-based optimization in this first submission or not.   Even though optimization is on our near-term roadmap, I'm leaning towards not including it in this first submission.  I think we should focus entirely on the parameterization framework itself.      That said, I do think the extra transformations discussed in Issues #115 and #89 should be included.   What do you think?

The current list of issues to close before JOSS submission would be #40, #44, #70, #71, #80, #82, #89, #106, #115, #124, #141 .   What do you all think?

**Comment by rubiop** [comment]
I am also going towards not including optimizing stuff. I think including optimization will also mean that will include distributions object etc. Even if I think that won't take too much time to reach this point I think it is good to stop earlier. I also think that doing that will be kind of beneficial to highlight that anyone can easily interact with the library at a relatively low level and plug already existing code/scripts in all languages.  

I also agree with the list of issues and corresponding feature, some of them (like #106 and #115) are new features compared to existing codes (python transport maps and matlab).

FYI there is a PR for #40

**Reply by rubiop** [reply]
I think we can include also #141 and #71 if possible for the list of issues needed to be included

**Reply by mparno** [reply]
Added to my list above.

**Reply by rubiop** [reply]
Added a "JOSS paper" milestone so we can update the list of issue more easily

**Comment by bonh** [comment]
But optimization is still utilized in the examples, right? Or are there some low-level examples you are thinking about?

**Reply by rubiop** [reply]
I think we need examples with optimization but I was thinking that optimization could be done using packages in the host language (like in the examples where we use scipy.optimize for python and fminunc for matlab)

**Reply by bonh** [reply]
Okay, sounds good!

**Comment by bonh** [comment]
What about #140?

**Reply by rubiop** [reply]
Added it to the "JOSS paper" milestone
